worker-websocket-durable-objects/src/durable-object.ts

```diff
@@ -0,0 +1,145 @@
+    const [client, server] = Object.values(pair);
+
+    // We're going to take pair[1] as our end, and return pair[0] to the client.
+    await this.handleWebSocketSession(server, requestMetadata);
+
+    // Now we return the other end of the pair to the client.
+    return new Response(null, { status: 101, webSocket: client });
+  }
+
+  async handleWebSocketSession(webSocket: WebSocket, metadata: IncomingRequestCfProperties) {
+    // Accept our end of the WebSocket. This tells the runtime that we'll be terminating the
+    // WebSocket in JavaScript, not sending it elsewhere.
+    // @ts-ignore
+    webSocket.accept();
+
+    // Create our session and add it to the users map.
+    const userId = crypto.randomUUID();
+    this.users.set(userId, {
+      id: userId,
+      city: metadata.city,
+      country: metadata.country,
+      websocket: webSocket,
+    });
+
+    webSocket.addEventListener('message', async msg => {
+      try {
+        // Parse the incoming message
+        let incomingMessage = <Message & MessageData>JSON.parse(msg.data);
+
+        switch (incomingMessage.type) {
+          case 'ping':
+            const msg = {
+              type: 'pong',
+              data: {
+                id: incomingMessage.data.id,
+                time: Date.now(),
+                dolocation: this.dolocation,
+                users: Array.from(this.users.values()).map(x => {
+                  // update user's ping
+                  if (incomingMessage.data.lastPingMs && x.websocket === webSocket) {
+                    this.pings.set(x.id, incomingMessage.data.lastPingMs);
+                  }
+
+                  return {
+                    ...x,
+                    ping: this.pings.get(x.id),
+                    websocket: undefined,
+                  };
+                }),
+              },
+            };
+            webSocket.send(JSON.stringify([msg]));
+            break;
+        }
+      } catch (err) {
+        // Report any exceptions directly back to the client. As with our handleErrors() this
+        // probably isn't what you'd want to do in production, but it's convenient when testing.
+        webSocket.send(JSON.stringify({ error: err.stack }));
+      }
+    });
+
+    // On "close" and "error" events, remove the WebSocket from the webSockets list
+    let closeOrErrorHandler = () => {
+      console.log('user', userId);
+      this.users.delete(userId);
+    };
+    webSocket.addEventListener('close', closeOrErrorHandler);
+    webSocket.addEventListener('error', closeOrErrorHandler);
+  }
+
+  // broadcast() broadcasts a message to all clients.
+  broadcast(message: string) {
+    // Iterate over all the sessions sending them messages.
+    this.users.forEach((user, key) => {
+      try {
+        user.websocket.send(message);
+      } catch (err) {
+        this.users.delete(key);
+      }
+    });
+  }
+
+  async getDurableObjectLocation() {
+    const res = await fetch('https://request.eidam.cf');
```

Comment:
let's find another way to do this. this makes the example reliant on an external, non-CF property which I know is yours (and i trust it/you) but it may not be there forever. examples should be 100% self-reliant imo

Reply:
added CORS headers needed and used https://workers.cloudflare.com/cf.json (the one Miniflare is using)